### DISCUSSION TITLE
GitHub Action for SonarQube code scanning without a dedicated hosted server

### DISCUSSION THREAD

**Discussion by gitricko** [community/community#129878] [Actions]
### Sonarless

SonarQube is a staple for many large companies due to its long-standing reputation in code coverage and SAST (Static Application Security Testing). Its free community version is powerful but requires hosting, which means engineers must commit code to get it scanned. This often ties the process to CI setups like Jenkins or GitHub Actions, resulting in unnecessary commits and extra overhead.

For engineers, this can be cumbersome and inefficient. To solve this, I've developed **_Sonarless_** – a CLI tool that eliminates the need for a centrally hosted SonarQube instance. Sonarless is designed to simplify your workflow by:

### Reducing Infrastructure Overhead: 
No need for a dedicated SonarQube server. 
Simple installation via `curl -s "https://raw.githubusercontent.com/gitricko/sonarless/main/install.sh" | bash'

### Streamlining Code Scanning: 
Run `sonarless scan` directly from your git root without waiting for CI processes.

### Seamless CI Integration: 
Includes its own GitHub Action, allowing you to incorporate code scanning into your CI pipeline effortlessly. All current SonarQube Scan GH Actions needs you to point a dedicated server.

### Cross-Platform Compatibility: 
Works on both Linux and Mac.

With Sonarless, engineers can achieve efficient code scanning without the hassle of managing additional infrastructure.

Check it out: [Sonarless GitHub Repository](https://github.com/gitricko/sonarless). If you find it useful, please give it a star and share your feedback!

**Comment by crispopp** [comment]
SonarQube is widely trusted by large companies for its robust code coverage and SAST capabilities. However, its community version, while powerful, requires hosting, tying engineers to commit code for scanning, often through CI tools like Jenkins or GitHub Actions, leading to unnecessary commits and added complexity.

To address this, I've developed Sonarless – a CLI tool that eliminates the need for a centrally hosted SonarQube instance. With **[Capra Host](https://caprahost.com/)** in mind, Sonarless simplifies your workflow by:

Reducing Infrastructure Overhead:
No dedicated SonarQube server needed.
Simple installation via curl -s "https://raw.githubusercontent.com/gitricko/sonarless/main/install.sh" | bash.

Streamlining Code Scanning:
Execute sonarless scan directly from your git root, bypassing CI wait times.

Seamless CI Integration:
Includes its own GitHub Action, seamlessly integrating code scanning into your CI pipeline without requiring a dedicated server.

Cross-Platform Compatibility:
Compatible with both Linux and Mac operating systems.

Sonarless empowers engineers to conduct efficient code scanning with Capra Host, eliminating the complexities of managing additional infrastructure.

Explore it here: [Sonarless GitHub Repository](https://github.com/gitricko/sonarless). If you find it valuable, please star the repository and share your feedback!
